### PR TITLE
RHS HIDF Template

### PR DESCRIPTION
**Pull request by ante185**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Adds the RHS HIDF faction.
This template challenges players to a fight against an foe with an outdated cold war era army, but friends in high places allow them some aces up their sleeve.

this also adds a config fix for the RHS M3A1 gun, as its lack of AI firemodes restricted its usefulness to none

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:
Dependencies: 
This template depends on #3172 as it allows their CAS plane to work.

### CODE REVIEW

File: A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf [hunk 1]
```diff
@@ -0,0 +1,856 @@
...
+["vehiclesPlanesAA", ["rhs_l159_cdf_b_CDF_CAP"]] call _fnc_saveToTemplate;
+["vehiclesPlanesTransport", []] call _fnc_saveToTemplate;
+["vehiclesAirPatrol", ["rhs_uh1h_hidf", "rhsgred_hidf_cessna_o3a"]] call _fnc_saveToTemplate;
+
+["vehiclesHelisLight", ["rhs_uh1h_hidf_unarmed"]] call _fnc_saveToTemplate;
+["vehiclesHelisTransport", ["rhs_uh1h_hidf"]] call _fnc_saveToTemplate;
+["vehiclesHelisLightAttack", ["rhs_uh1h_hidf_gunship"]] call _fnc_saveToTemplate; 
+["vehiclesHelisAttack", []] call _fnc_saveToTemplate;
+
+["vehiclesArtillery", ["RHS_M119_WD"]] call _fnc_saveToTemplate;        // "rhsusf_M142_usarmy_D" - OP until we can customize magazines
```
Comment (Tiny-DM) [inline A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:50]: remove comment

File: A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf [hunk 2]
```diff
@@ -0,0 +1,856 @@
...
+["magazines", createHashMapFromArray [
+["rhsusf_m109_usarmy", ["rhs_mag_155mm_m795_28"]],
```
Comment (Tiny-DM) [inline A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:52]: remove unnecessary ammo entries

File: A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf [hunk 3]
```diff
@@ -0,0 +1,856 @@
...
+["rhsusf_M142_usarmy_WD", ["rhs_mag_m26a1_6", "rhs_mag_mgm168_block4_1", "rhs_mag_mgm140b_1"]],//Cluster, 500lb HE, 69 Small HE Rockets
+["RHS_M119_WD", ["RHS_mag_m1_he_12"]]
+]] call _fnc_saveToTemplate;
+
+["uavsAttack", []] call _fnc_saveToTemplate;    
+["uavsPortable", []] call _fnc_saveToTemplate;
+
+//Config special vehicles
+["vehiclesMilitiaLightArmed", ["rhsgref_hidf_m1025_m2"]] call _fnc_saveToTemplate;
+["vehiclesMilitiaTrucks", ["rhs_zil131_msv"]] call _fnc_saveToTemplate;
```
Comment (Tiny-DM) [inline A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:62]: Eastern truck stands out in a template of US surplus

File: A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf [hunk 4]
```diff
@@ -0,0 +1,856 @@
...
+_loadoutData set ["carbines", []];
+_loadoutData set ["grenadeLaunchers", []];
+_loadoutData set ["gltube", [["rhs_weap_m79", "", "", "",["rhs_mag_M441_HE","rhs_mag_m714_White","rhs_mag_m662_red"], [], ""]]];
+_loadoutData set ["SMGs", []];
+_loadoutData set ["machineGuns", []];
+_loadoutData set ["marksmanRifles", []];
+_loadoutData set ["sniperRifles", []];
+
+_loadoutData set ["lightATLaunchers", ["rhs_weap_m72a7"]];
+_loadoutData set ["ATLaunchers", [
```
Comment (Tiny-DM) [inline A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:108]: Less optics maybe? The faction is supposed to be really poor.

File: A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+["rhs_weap_m4a1", "rhsusf_acc_rotex5_grey", "acc_flashlight", "rhsusf_acc_compm4", 	["rhs_mag_30Rnd_556x45_M855A1_Stanag"], [], ""],
+["rhs_weap_m4a1_carryhandle", "rhsusf_acc_rotex5_grey", "acc_flashlight", "", 		["rhs_mag_30Rnd_556x45_M855A1_Stanag"], [], ""]
+]];
+_sfLoadoutData set ["grenadeLaunchers", [
+["rhs_weap_m4a1_carryhandle_m203", "rhsusf_acc_rotex5_grey", "", "rhsusf_acc_compm4",	["rhs_mag_30Rnd_556x45_M855A1_Stanag"], ["rhs_mag_M433_HEDP","rhs_mag_m714_White","rhs_mag_m662_red"], ""],
+["rhs_weap_m4a1_carryhandle_m203", "rhsusf_acc_rotex5_grey", "", "rhsusf_acc_compm4",	["rhs_mag_30Rnd_556x45_M855A1_Stanag"], ["rhs_mag_M397_HET","rhs_mag_m714_White","rhs_mag_m662_red"], ""]
+]];
+_sfLoadoutData set ["SMGs", [
+["rhs_weap_m3a1_specops", "", "", "rhsusf_acc_compm4", ["rhsgref_30rnd_1143x23_M1911B_SMG"], [], ""],
+["rhs_weap_m4a1_carryhandle", "rhsusf_acc_rotex5_grey", "acc_flashlight", "", ["rhs_mag_30Rnd_556x45_M855A1_Stanag"], [], ""]
```
Comment (Tiny-DM) [inline A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:206]: An M4 in the SMGs?

**Review by Tiny-DM — CHANGES_REQUESTED** [review]

File: A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf (hunk 5 above)
Comment (ante185) [reply A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:204]: yep

**Review by Tiny-DM — APPROVED** [review]
lgtm

**Comment by ante185** [comment]
Update:
Replaced the modern US trucks with trucks of a more suitable vintage.
Created olive green versions of the TOW hmmwv, m113 m240, m113 MEV, and M2A2, added to template

**Pull request merged**